src/range.rs

```diff
@@ -203,22 +203,55 @@ impl<V: Ord> Range<V> {
     pub fn contains(&self, v: &V) -> bool {
         for segment in self.segments.iter() {
             if match segment {
-                (Unbounded, Unbounded) => true,
-                (Unbounded, Included(end)) => v <= end,
-                (Unbounded, Excluded(end)) => v < end,
-                (Included(start), Unbounded) => v >= start,
-                (Included(start), Included(end)) => v >= start && v <= end,
-                (Included(start), Excluded(end)) => v >= start && v < end,
-                (Excluded(start), Unbounded) => v > start,
-                (Excluded(start), Included(end)) => v > start && v <= end,
-                (Excluded(start), Excluded(end)) => v > start && v < end,
+                (Excluded(start), _) => v <= start,
+                (Included(start), _) => v < start,
+                (Unbounded, _) => false,
```

Comment:
Isn't this wrong if the very first segment is `(Unbounded, Unbounded)`?

Reply:
This whole short-circuit change feels shady ...

Reply:
The `false` here means that it fails to short-circuit and falls through to the check below which correctly returns `true`. That being said I need to reread it twice to figure out what was going on here, and I wrote the code yesterday. There has to be a clear way to write this.

Similarly, we can probably improve the testing to ensure that the code are correct, even when were not paying attention. I will need to look into what is worth the effort. (One day, I would love to use kani or creusot to prove the code correct, not that I have time for it now.)